numpy/lib/npyio.py

```diff
@@ -519,23 +515,18 @@ def save(file, arr, allow_pickle=True, fix_imports=True):
     >>> print(a, b)
     # [1 2] [1 3]
     """
-    own_fid = False
-    if hasattr(file, 'write'):
-        fid = file
-    else:
-        file = os_fspath(file)
-        if not file.endswith('.npy'):
-            file = file + '.npy'
-        fid = open(file, "wb")
-        own_fid = True
+    with contextlib.ExitStack() as stack:
+        if hasattr(file, 'write'):
+            fid = file
+        else:
+            file = os_fspath(file)
+            if not file.endswith('.npy'):
+                file = file + '.npy'
+            fid = stack.enter_context(open(file, "wb"))
```

Comment:
This one might be a better candidate for `contextlib.nullcontext`, which we use elsewhere - as:
```python
if hasattr(file, 'write'):
    fctx = contextlib_nullcontext(file)
else:
    file = os_fspath(file)
    if not file.endswith('.npy'):
        file = file + '.npy'
    fctx=open(file, "wb")
with fctx:
   ...
```

Reply:
I don't believe this will work because you need a way to transfer ownership to NpzFile (and not close the context).

Reply:
I thought only one of these two cases did that. Let me reread...

Reply:
I don't see any use of `pop_all()` for this section

Reply:
Sorry I thought this comment was attached to the upper section, your suggestion is a good improvement.

Thanks for all the through (and timely) reviews you've been giving.